pkg/daemon/ceph/osd/kms/kms.go

```diff
@@ -114,6 +117,33 @@ func (c *Config) PutSecret(secretName, secretValue string) error {
 			return errors.Wrap(err, "failed to put secret in ibm key protect")
 		}
 	}
+	if c.IsKMIP() {
+		_, err := c.getKubernetesSecret(secretName)
+		if err == nil {
+			// if error is nil, secret exists, just return nil.
+			return nil
+		}
+		// if error is not found, continue with creation.
+		if !kerrors.IsNotFound(err) {
```

Comment:
if its already exist also we can proceed, isnt it ?

Reply:
> if its already exist also we can proceed, isnt it ?

no, that will create an extra key. 
if the secret with unique id is already created, we can just return.